### PR TITLE
feat(Table): Add column reordering

### PR DESCRIPTION
**Pull request by FlyersPh9**
Added to support [feat(Table): support for draggable columns #561](https://github.com/iTwin/iTwinUI-react/pull/561).

- Add `.iui-reorder-column-left` & `.iui-reorder-column-right` classes to give a border to show where the column is being placed.
Screen Shot 2022-02-25 at 11 45 23 AM
- Add `.iui-grabbing` modifier to give `cursor: grabbing;` when the column is being dragged (similar to slider).

### CODE REVIEW

File: src/table/table.scss [hunk 1]
```diff
@@ -99,6 +103,12 @@
     }
   }
 
+  .iui-reorder-column {
+    @include themed {
+      border-right: $iui-xxs solid t(iui-color-foreground-primary);
```
Comment (veekeys) [inline src/table/table.scss:108]: Not sure if its always right though

https://github.com/iTwin/iTwinUI-react/pull/561/files#r814875116

File: src/table/table.scss [hunk 2]
```diff
@@ -40,6 +40,10 @@
       cursor: pointer;
     }
 
+    &.iui-grabbing {
+      cursor: grabbing;
```
Comment (veekeys) [inline src/table/table.scss:44]: You might also want to indicate somehow, that column is available for reorder.
`cursor: grab;`, again similar to slider

File: src/table/table.scss (hunk 1 above)
Comment (FlyersPh9) [reply src/table/table.scss:108]: Updated to work on left or right.

File: src/table/table.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/table/table.scss:44]: I thought about that, but the entire column header is clickable and is using `cursor: pointer;`.

Comment (mayank99) [reply src/table/table.scss:44]: Maybe a light border on hover?

Comment (veekeys) [reply src/table/table.scss:44]: but wont it be weird, when cursor from pointer will change to grabbing ?

**Review by veekeys — APPROVED** [review]
If the UI is fine, for initial version we can have just this, so PR in react would not be stopped by this.

File: src/table/table.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/table/table.scss:44]: At least on hover the cursor changes to the pointer hand, so its not a weird transition from `default` to hand.

**Pull request merged**